guardrails-scripts/yb-voyager-pg-grant-migration-permissions.sql

```diff
@@ -165,7 +156,31 @@ GRANT pg_read_all_stats to :voyager_user;
     -- Add the original owner of the tables to the group
     \echo ''
     \echo '--- Adding Original Owner to Replication Group ---'
-    GRANT :replication_group TO :original_owner_of_tables;
+    DO $$
+    DECLARE
+        tableowner TEXT;
+        schema_list TEXT[] := string_to_array(current_setting('myvars.schema_list'), ',');  -- Convert the schema list to an array
+        replication_group TEXT := current_setting('myvars.replication_group');  -- Get the replication group from settings
+    BEGIN
+        -- Generate the GRANT statements and execute them dynamically
+        FOR tableowner IN
+            SELECT DISTINCT t.tableowner
+            FROM pg_catalog.pg_tables t
+            WHERE t.schemaname = ANY (schema_list)  -- Use the schema_list variable
+            AND NOT EXISTS (
+                SELECT 1
+                FROM pg_roles r
+                WHERE r.rolname = t.tableowner
+                    AND pg_has_role(t.tableowner, replication_group, 'USAGE')  -- Use the replication_group variable
+            )
```

Comment:
here we are ignoring if tableowner is already a part of replication group, is that correct?

Reply:
got it